emscripten.py

```diff
@@ -498,11 +504,38 @@ def finalize_wasm(infile, outfile, memfile, js_syms):
   if settings.DEBUG_LEVEL >= 3:
     args.append('--dwarf')
 
-  metadata = get_metadata(infile, outfile, modify_wasm, args)
+  if settings.GENERATE_SOURCE_MAP:
+    building.emit_wasm_source_map(infile, infile + '.map', outfile)
+    building.save_intermediate(infile + '.map', 'base_wasm.map')
+    base_url = settings.SOURCE_MAP_BASE + os.path.basename(outfile) + '.map'
+    if modify_wasm:
+      # If we are already modifying, just let Binaryen add the sourcemap URL
+      args += ['--output-source-map-url=' + base_url]
+    else:
+      # Otherwise use objcopy. This avoids re-encoding the file (thus
+      # preserving DWARF) and is faster.
+      url_file = create_sourcemapping_url_file(infile, base_url)
+      cmd = [shared.LLVM_OBJCOPY,
+             '--add-section',
+             'sourceMappingURL=' + url_file,
+             infile]
+      shared.check_call(cmd)
+
+  post_strip_file = infile
+  if not settings.GENERATE_DWARF or not settings.EMIT_PRODUCERS_SECTION:
+    # For sections we no longer need, strip now to speed subsequent passes
+    post_strip_file = shared.get_temp_files().get('.wasm').name
+    building.save_intermediate(infile, 'strip.wasm')
+    sections = ['producers'] if not settings.EMIT_PRODUCERS_SECTION else []
+    building.strip(infile, post_strip_file, debug=not settings.GENERATE_DWARF,
+                   sections=sections)
+
+  metadata = get_metadata(post_strip_file, outfile, modify_wasm, args)
   if modify_wasm:
     building.save_intermediate(infile, 'post_finalize.wasm')
```

Comment:
```suggestion
    building.save_intermediate(post_strip_file, 'post_finalize.wasm')
```
Shouldn't this be changed too?

Reply:
Yes. But see below